azure-proto-compute/AvailabilitySetContainer.cs

```diff
@@ -9,30 +9,40 @@
 namespace azure_proto_compute
 {
     /// <summary>
-    /// Operations class for Availability Set Contaienrs (resource groups)
+    /// A class representing collection of availability set and their operations over a resource group.
     /// </summary>
     public class AvailabilitySetContainer : ResourceContainerBase<AvailabilitySet, AvailabilitySetData>
     {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="AvailabilitySetContainer"/> class.
+        /// </summary>
+        /// <param name="options"> The client parameters to use in these operations. </param>
+        /// <param name="resourceGroup"> The data of Resource Group. </param>
         internal AvailabilitySetContainer(AzureResourceManagerClientOptions options, ResourceGroupData resourceGroup)
             : base(options, resourceGroup)
         {
         }
 
+        /// <summary>
+        /// Initializes a new instance of the <see cref="AvailabilitySetContainer"/> class.
+        /// </summary>
+        /// <param name="options"> The client parameters to use in these operations. </param>
+        /// <param name="id"> The identifier of the resource that is the target of operations. </param>
         internal AvailabilitySetContainer(AzureResourceManagerClientOptions options, ResourceIdentifier id)
             : base(options, id)
         {
         }
 
-        protected override ResourceType ValidResourceType => ResourceGroupOperations.ResourceType;
-
-        public override ArmResponse<AvailabilitySet> Create(string name, AvailabilitySetData resourceDetails)
+        /// <inheritdoc/>
+        public override ArmResponse<AvailabilitySet> Create(string name, AvailabilitySetData resourceDetails, CancellationToken cancellationToken = default)
         {
             var response = Operations.CreateOrUpdate(Id.ResourceGroup, name, resourceDetails.Model);
             return new PhArmResponse<AvailabilitySet, Azure.ResourceManager.Compute.Models.AvailabilitySet>(
                 response,
                 a => new AvailabilitySet(ClientOptions, new AvailabilitySetData(a)));
         }
 
+        /// <inheritdoc/>
```

Comment:
Not sure we should inherit here since you lose context of the type.  The base only says TOperations which doesn't make it clear that you are getting back an AvailabilitySet object